drift-client/src/main/java/io/airlift/drift/client/DriftMethodHandler.java

```diff
@@ -62,6 +73,20 @@ public boolean isAsync()
 
     public ListenableFuture<Object> invoke(Optional<String> addressSelectionContext, Map<String, String> headers, List<Object> parameters)
     {
+        if (!headerParameters.isEmpty()) {
+            headers = new LinkedHashMap<>(headers);
+            for (Entry<Integer, ThriftHeaderParameter> entry : headerParameters.entrySet()) {
+                headers.put(entry.getValue().getName(), (String) parameters.get(entry.getKey()));
```

Comment:
Why is `parameters` not `List<String>`?

Reply:
Parameters are the parameters to the method being invoked... the arguments.  In the case of a header it is required to be a `String` but for `@Thrift` parameters they can be anything